tools/lib/node.coffee

```diff
@@ -0,0 +1,62 @@
+semver = require('semver')
+utils = require('./utils')
+
+RESIN_URL_PREFIX = 'http://resin-packages.s3.amazonaws.com/node/'
+NODEJS_URL_PREFIX = 'http://nodejs.org/dist/'
+
+UNSUPPORTED = 'UNSUPPORTED'
+
+ALPINE_ARCH =
+	'amd64': 'alpine-amd64'
+	'i386': 'alpine-i386'
+	'armv6hf': 'alpine-armhf'
+	'armv7hf': 'alpine-armhf'
```

Comment:
So, it should be working now. [https://github.com/resin-io-playground/metalsmith-dynamic#glossary]

Update to `metalsmith-dynamic@^0.4.0`
Create a new dictionary called `arch.json`. As the very minimum it should contain an array with objects `{ "id": "armvh7" }`, or the like.

Now in the device types dictionary switch from string arch refs to the new FKs:
from `"architecture": "armvh7"` to `"$architecture": "$arch.armvh7"`

Obviously in the source files you have to switch from `$device_type.architecture` to `$device_type.$architecture.id`.

Now you can add arbitrary props to the arch dictionary and use them in the HB helpers.